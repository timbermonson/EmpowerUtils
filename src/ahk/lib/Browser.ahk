#Requires AutoHotkey v2.0

#Include etc.ahk

class Browser {
    static clipboardError := "$clipboardError"

    static clipboardInjector :=
        "function ctc(text) {{}    const input = document.createElement('input');    input.value = text;    document.body.appendChild(input);    input.select();    document.execCommand('copy');    document.body.removeChild(input);{}}"

    static jQueryInjector :=
        "var script = document.createElement('script'); script.src = `"https://code.jquery.com/jquery-3.7.1.min.js`"; document.getElementsByTagName('head')[0].appendChild(script);"

    static toggleConsole(wait := 600) {
        Send "{Ctrl down}{Shift down}j{Shift up}{Ctrl up}"
        Sleep wait
    }

    static setupConsole(initialToggle := true) {
        if (initialToggle) {
            this.toggleConsole(300)
        }

        winW := 0
        winH := 0
        WinGetPos(, , &winW, &winH, "A")

        if (!ScanCursor(60, winH - 30, "U", 800)) {
            this.toggleConsole(300)
            if (!ScanCursor(60, winH - 30, "U", 800)) {
                throw TargetError()
            }
        }

        Send("{LButton down}")
        MouseMove(60, winH - 165)
        Sleep 50

        Send("{LButton up}")
        Sleep 50

        MouseMove(60, winH - 60)
        Send "{LButton 2}"
    }

    static cmd(cmd, fromClipboard := true, useClipboardForErrors := true) {

        if (fromClipboard) {
            if (useClipboardForErrors) {
                cmd := "try{" . cmd . "}catch(e){ctc(`"" . this.clipboardError . "`")}"
            }
            A_Clipboard := cmd
            Send "^v"
        } else {
            SetKeyDelay(0)

            SendEvent cmd
            SendEvent "{Escape 2}"
        }
        Sleep 75
        Send "{Enter 3}"
        Sleep 75
    }

    static cmdToClipboard(cmd, useClipboard := true) {
        cmd := "ctc(" . cmd . ")"
        this.cmd(cmd, useClipboard, useClipboard)
    }

    static setupFunctions() {
        A_Clipboard := ""
        while (A_Clipboard != "$clipboardConfirm") {
            this.cmd(this.clipboardInjector, , false)
            this.cmdToClipboard("`"$clipboardConfirm`"", false)
        }

        this.cmd("allow pasting", false, false)
        this.cmd("allow pasting", false, false)

        A_Clipboard := ""
        while (A_Clipboard != "function") {
            this.cmd(this.jQueryInjector)
            this.cmdToClipboard("typeof jQuery")
        }
    }

    static setup() {
        this.setupConsole()
        this.setupFunctions()
    }

    static waitForPageChange(timeout := 5000) {
        this.cmdToClipboard("`"$clipboardConfirm`"", false)
        this.cmdToClipboard("`"$clipboardConfirm`"", false)
        start := A_TickCount

        while (A_TickCount - start <= timeout) {
            A_Clipboard := ""

            this.cmdToClipboard("`"$clipboardConfirm`"", false)
            this.cmdToClipboard("`"$clipboardConfirm`"", false)

            if (A_Clipboard != "`"$clipboardConfirm`"") {
                this.setupFunctions()
                return
            }
        }

        throw TargetError()
    }

    static toQuery(q, cmd := "") {
        return this.toQueryPlain(q) . ".get(0)" . cmd
    }

    static toQueryPlain(q, cmd := "") {
        replacedQ := RegExReplace(esc(q), "i)\^h\[([^]]+)\]", "`").has(`"$1`").find(`"")

        return fullQuery := "jQuery(`"" . replacedQ . "`")" . cmd
    }

    static copyQ(q) {
        this.cmdToClipboard(this.toQuery(q, ".textContent"))
        if (A_Clipboard = this.clipboardError) {
            throw TargetError()
        }
    }

    static waitClickQ(qList) {
        if (!IsObject(qList)) {
            qList := [qList]
        }

        return this.waitForQ(qList, , , this.clickQ)
    }

    static existsQ(q) {
        Browser.cmdToClipboard(Browser.toQueryPlain(q,
            ".length"))

        if (A_Clipboard = "1") {
            return 1
        }
    }

    static clickQ(q) {
        this.cmd(this.toQuery(q, ".click()"))
        if (A_Clipboard = this.clipboardError) {
            throw TargetError()
        }
    }

    static typeQ(q, input) {
        this.cmd(this.toQuery(q, ".value = `"") . input . "`"")

        this.cmd(this.toQuery(q, ".dispatchEvent(new KeyboardEvent(`"keyup`"))"))
        this.cmd(this.toQuery(q, ".dispatchEvent(new KeyboardEvent(`"keyup`"))"))
    }

    static waitForQ(qList, timeout := 8000, interval := 100, callback := -1, callbackParam := -1) {
        start := A_TickCount

        if (!IsObject(qList)) {
            qList := [qList]
        }

        while (A_TickCount - start <= timeout) {
            for index, q in qList {
                Browser.cmdToClipboard(Browser.toQueryPlain(q,
                    ".length"))

                if (A_Clipboard = "1") {
                    if (callback != -1) {
                        if (callbackParam = -1) {
                            callbackParam := q
                        }
                        callback.Call(this, callbackParam)
                    }
                    return index
                }
            }
            sleep interval
        }

        throw TargetError()
    }
}
