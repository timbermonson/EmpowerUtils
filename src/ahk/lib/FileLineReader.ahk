#Requires AutoHotkey v2.0

#Include etc.ahk

class FileLineReader {
    filePath := ""
    lineList := []
    firstLine := ""

    __New(filePath) {
        this.filePath := filePath
        this.readInput()
    }

    readInput() {
        fileContents := FileRead(this.filePath, "UTF-8")

        this.lineList := StrSplit(fileContents, "`n")
        if (this.lineList.length < 2) {
            MsgBox("AAA")
            return
        }
        this.firstLine := Trim(this.lineList[1])
    }

    removeFirstLine() {
        if (this.lineList.length < 1) {
            return
        }

        FileDelete(this.filePath)
        this.lineList.RemoveAt(1)
        FileAppend(StrJoin(this.lineList, "`n"), this.filePath)
        Sleep 100
        this.readInput()
    }
}
