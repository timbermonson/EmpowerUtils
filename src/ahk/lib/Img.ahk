#Requires AutoHotkey v2.0

class Img {
    static waitForImages(imageNameList, interval := 200, timeout := 2000) {
        start := A_TickCount

        while (A_TickCount - start <= timeout) {
            for ind, imageName in imageNameList {
                if (this.hasImage(imageName) = 1)
                    return ind
            }

            Sleep interval
        }

        return -1
    }

    static hasImage(imageName, xBound := 2000, yBound := 2000) {
        x := -1
        y := -1
        return ImageSearch(&x, &y, 0, 0, xBound, yBound, "*n100 " . A_ScriptDir . "/images/" . imageName)
    }

    static searchClickImage(imageName, xOffset := 0, yOffset := 0, xBound := 2000, yBound := 2000) {
        x := -1
        y := -1
        found := ImageSearch(&x, &y, 0, 0, xBound, yBound, "*n100 " . A_ScriptDir . "/images/" . imageName)
        if (!found) {
            MsgBox("Not found!")
            return false
        }

        MouseMove x + xOffset, y + yOffset
        MouseClick
        return true
    }
}
