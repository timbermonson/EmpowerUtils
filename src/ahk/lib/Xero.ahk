#Requires AutoHotkey v2.0

#Include Browser.ahk
#Include etc.ahk

class Xero {
    static dashOpAccBtn :=
        "div.mf-bank-widget-panel^h[h2:contains(`"Operating:`"):contains(`"2894`")].mf-bank-widget-touchtarget"
    static orgSearchNone := ".xnav-orgsearch--message:contains(`"No results found.`")"
    static orgSearchFirst := "ol[role=`"navigation`"].xnav-verticalmenu > li:nth-child(1) > a"

    static setup() {
        Browser.setup()
    }

    static switchToOrg(orgName) {
        Browser.setupConsole(false)

        if (!Browser.existsQ(".xnav-appmenu--body-is-open")) {
            Browser.clickQ(".xnav-appbutton--body")
        }

        if (Browser.existsQ(".xnav-orgsearch--input")) {
            Browser.clickQ(".xnav-icon-orgsearchclear")
        }

        Browser.waitClickQ("[data-name=`"xnav-changeorgbutton`"]")
        Browser.typeQ("[title=`"Search organizations`"]", orgName)
        Browser.waitForQ(".xnav-loader-wrapper")
        if (Browser.waitForQ([this.orgSearchNone, this.orgSearchFirst]) = 1) {
            throw Error("No results for " . orgName)
        }

        Browser.waitClickQ(this.orgSearchFirst)
        Browser.waitForPageChange()

        Browser.waitForQ(this.dashOpAccBtn)
    }

    static openOperatingImports() {
        Browser.clickQ(this.dashOpAccBtn)
        Browser.waitClickQ("a.mf-bank-widget-text-minorlink:contains(`"Import a Statement`")")
        Browser.waitForPageChange()
    }
}
