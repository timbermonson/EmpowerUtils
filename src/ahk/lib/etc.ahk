#Requires AutoHotkey v2.0

ScanCursor(xStart, yStart, dir, dist, cursorOverride := 0) {
    CoordMode("Mouse", "Window")
    stepDistance := 5
    curs := ""
    xStep := 0
    yStep := 0
    switch (StrUpper(dir)) {
        case "U":
            yStep := -stepDistance
            curs := "SizeNS"
        case "D":
            yStep := stepDistance
            curs := "SizeNS"
        case "L":
            xStep := -stepDistance
            curs := "SizeWE"
        case "R":
            xStep := stepDistance
            curs := "SizeWE"
        default:
            throw ValueError("ScanCursor: Invalid Direction!")
    }
    if (cursorOverride != 0) {
        curs := cursorOverride
    }

    x := xStart
    y := yStart

    loop dist / stepDistance {
        MouseMove(x, y)
        Sleep 1

        if (A_Cursor = curs) {
            return 1
        }
        x += xStep
        y += yStep
    }

    return
}

StrJoin(arr, sep) {
    if (arr.Length = 0) {
        return ""
    }

    result := arr[1]

    for ind, str in arr {
        if (ind = 1) {
            continue
        }

        result .= sep
        result .= str
    }

    return result
}

esc(str) {
    return StrReplace(str, "`"", "\`"")
}
