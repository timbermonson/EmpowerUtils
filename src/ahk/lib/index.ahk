#Requires AutoHotkey v2.0

#Include etc.ahk

#Include Browser.ahk
#Include FileLineReader.ahk
#Include Img.ahk
#Include Xero.ahk