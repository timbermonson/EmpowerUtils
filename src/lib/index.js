import * as address from './address.js'
import * as browser from './browser.js'
import * as io from './io.js'
import * as scraper from './scraper.js'
import * as Xero from './Xero.js'
import * as str from './string.js'

export default { io, address, scraper, str, browser, Xero }
