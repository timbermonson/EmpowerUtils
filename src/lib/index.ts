import * as address from './address.js'
import * as etc from './etc.js'
import * as io from './io.js'
import * as scraper from './scraper.js'
import * as str from './string.js'

import Xero from './Xero.js'
import AutoBrowser from './AutoBrowser.js'
import InputLineIterator from './InputLineIterator.js'

export default {
    address,
    AutoBrowser,
    etc,
    InputLineIterator,
    io,
    scraper,
    str,
    Xero,
}
