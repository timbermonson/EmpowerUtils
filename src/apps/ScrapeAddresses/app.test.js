import lib from '../../lib/index.ts'
vi.mock('../../lib/index.ts', { spy: true })

import * as localLib from './lib.js'
vi.mock('./lib.js')

import countyScraperMap from './countyPlugins/index.js'
vi.mock('./countyPlugins/index.js', () => {
    return { default: { countyA: vi.fn(), countyB: vi.fn() } }
})

import {
    getOutputText,
    getSearchresultMapByName,
    parseInputSingle,
    parseInputMultiple,
    run,
} from './app'

const testNameSearchResultMap = {
    'george washington': {
        fullName: 'georg washton',
        addressList: [
            { street: ' pick this ', city: ' city' },
            { street: " don't pick this ", city: 'bad city ' },
        ],
    },
    'john smith': {
        fullName: 'johnathan smithson',
        addressList: [
            { street: ' jpick this ', city: ' jcity' },
            { street: " jdon't pick this ", city: 'jbad city ' },
        ],
    },
}
const testSingleName = 'george washington'

describe('Address Scraper App', () => {
    beforeEach(() => {
        vi.clearAllMocks()
    })

    describe('parseInputSingle()', () => {
        test('Handles empty input', () => {
            expect(parseInputSingle('')).toStrictEqual([])
        })

        test('Trims empty lines', () => {
            expect(parseInputSingle('\n\n\n')).toStrictEqual([])
        })

        test('Cleans up messy input', () => {
            const inp =
                '   \t\r\n title            firstName  middle           Last    , a   B c\n'
            const out = parseInputSingle(inp)
            expect(out).toStrictEqual(['firstname middle last', 'b c'])
        })

        test('Ignores additional populated lines', () => {
            const inp = 'a b c, d e f\ng h i, j k l'
            const out = parseInputSingle(inp)
            expect(out).toStrictEqual(['b c', 'e f'])
        })
    })

    describe('parseInputMultiple()', () => {
        test("Doesn't trim empty lines", () => {
            const out = parseInputMultiple('\n\n')

            expect(out).toStrictEqual([[], [], []])
        })
        test('Parses properly, creating a list of namelists', () => {
            const inp =
                '   \t\r\n title            firstName  middle           Last    , a   B c\n\nbeeper Boop e Boop'
            const exp = [
                [],
                ['firstname middle last', 'b c'],
                [],
                ['boop e boop'],
            ]

            const out = parseInputMultiple(inp)
            expect(out).toStrictEqual(exp)
        })
    })

    describe('getOutputText()', () => {
        test('Throws on bad format setting', () => {
            expect(() =>
                getOutputText({}, { format: 'zzzbadformat' })
            ).toThrowError()
        })

        describe('Excel output', () => {
            test('Handles 0 names', () => {
                const out = getOutputText({}, { format: 'excel' })
                expect(out).toEqual('')
            })

            test('Handles 1 name', () => {
                const inpObj = {
                    [testSingleName]: testNameSearchResultMap[testSingleName],
                }

                const out = getOutputText(inpObj, { format: 'excel' })
                expect(out).toMatchSnapshot()
            })

            test('Handles 2 names', () => {
                const out = getOutputText(testNameSearchResultMap, {
                    format: 'excel',
                })
                expect(out).toMatchSnapshot()
            })
        })

        describe('JSON output', () => {
            test('Handles 0 names', () => {
                const out = getOutputText({}, { format: 'json' })
                expect(out).toEqual('{}')
            })

            test('Handles 1 name', () => {
                const inpObj = {
                    [testSingleName]: testNameSearchResultMap[testSingleName],
                }

                const out = getOutputText(inpObj, { format: 'json' })
                expect(out).toMatchSnapshot()
            })

            test('Handles 2 names', () => {
                const out = getOutputText(testNameSearchResultMap, {
                    format: 'json',
                })
                expect(out).toMatchSnapshot()
            })
        })
        describe('Both output', () => {
            test('Handles 0 names', () => {
                const out = getOutputText({}, { format: 'both' })
                expect(out).toEqual('{}\t')
            })

            test('Handles 1 name', () => {
                const inpObj = {
                    [testSingleName]: testNameSearchResultMap[testSingleName],
                }

                const out = getOutputText(inpObj, { format: 'both' })
                expect(out).toMatchSnapshot()
            })

            test('Handles 2 names', () => {
                const out = getOutputText(testNameSearchResultMap, {
                    format: 'both',
                })
                expect(out).toMatchSnapshot()
            })
        })
    })

    describe('getSearchresultMapByName()', () => {
        test('Handles 0 names', async () => {
            const res = await getSearchresultMapByName([])

            expect(countyScraperMap.countyA).toBeCalledTimes(0)
            expect(countyScraperMap.countyB).toBeCalledTimes(0)

            expect(localLib.pickBestCountyAndAddresses).toBeCalledTimes(1)
            expect(localLib.pickBestCountyAndAddresses).toBeCalledWith({
                countyA: {},
                countyB: {},
            })
        })

        test('Handles 1 names', async () => {
            await getSearchresultMapByName(['george washington'])

            expect(countyScraperMap.countyA.mock.calls).toEqual([
                ['george washington'],
            ])
            expect(countyScraperMap.countyB.mock.calls).toEqual([
                ['george washington'],
            ])

            expect(
                localLib.pickBestCountyAndAddresses.mock.calls
            ).toMatchSnapshot()
        })

        test('Handles 2 names', async () => {
            const res = await getSearchresultMapByName([
                'george washington',
                'john smith',
            ])

            expect(countyScraperMap.countyA.mock.calls).toEqual([
                ['george washington'],
                ['john smith'],
            ])
            expect(countyScraperMap.countyB.mock.calls).toEqual([
                ['george washington'],
                ['john smith'],
            ])

            expect(
                localLib.pickBestCountyAndAddresses.mock.calls
            ).toMatchSnapshot()
        })
    })

    describe('run()', () => {
        const testSingleInput = 'D ab cd'
        const testMultiInput = `${testSingleInput}, D de fg\nD ifsingle donotread`
        const testAddressPickerReturn = {
            a: {
                fullName: 'useThisNameA',
                addressList: [{ city: 'Acity', street: 'Astreet' }],
            },
            b: {
                fullName: 'useThisNameB',
                addressList: [{ city: 'Bcity', street: 'Bstreet' }],
            },
        }

        describe('Single Line', () => {
            test('Handles 0 names', async () => {
                lib.io.getInputData.mockReturnValueOnce('')

                await run()

                expect(countyScraperMap.countyA.mock.calls).toEqual([])
                expect(countyScraperMap.countyB.mock.calls).toEqual([])
                expect(lib.io.appendOutputData).toBeCalledTimes(1)
            })

            test('Handles 1 name', async () => {
                lib.io.getInputData.mockReturnValueOnce(testSingleInput)

                await run()

                expect(countyScraperMap.countyA.mock.calls).toEqual([['ab cd']])
                expect(countyScraperMap.countyB.mock.calls).toEqual([['ab cd']])
                expect(lib.io.appendOutputData).toBeCalledTimes(1)
            })

            test('Handles 2 names, uses formatter', async () => {
                lib.io.getInputData.mockReturnValueOnce(testMultiInput)
                localLib.pickBestCountyAndAddresses.mockReturnValueOnce(
                    testAddressPickerReturn
                )

                await run()

                expect(countyScraperMap.countyA.mock.calls).toEqual([
                    ['ab cd'],
                    ['de fg'],
                ])
                expect(countyScraperMap.countyB.mock.calls).toEqual([
                    ['ab cd'],
                    ['de fg'],
                ])

                expect(lib.io.appendOutputData.mock.calls).toMatchSnapshot()
            })
        })

        describe('Multi Line', () => {
            test('Handles 0 names', async () => {
                lib.io.commandLineArgsWrapper.mockReturnValueOnce({
                    multiple: true,
                })
                lib.io.getInputData.mockReturnValueOnce('')

                await run()

                expect(countyScraperMap.countyA.mock.calls).toEqual([])
                expect(countyScraperMap.countyB.mock.calls).toEqual([])
                expect(lib.io.appendOutputData).toBeCalledTimes(1)
            })

            test('Handles 1 name', async () => {
                lib.io.commandLineArgsWrapper.mockReturnValueOnce({
                    multiple: true,
                })
                lib.io.getInputData.mockReturnValueOnce(testSingleInput)

                await run()

                expect(countyScraperMap.countyA.mock.calls).toEqual([['ab cd']])
                expect(countyScraperMap.countyB.mock.calls).toEqual([['ab cd']])
                expect(lib.io.appendOutputData).toBeCalledTimes(1)
            })

            test('Handles 2+1 names', async () => {
                lib.io.commandLineArgsWrapper.mockReturnValueOnce({
                    multiple: true,
                })
                lib.io.getInputData.mockReturnValueOnce(testMultiInput)

                await run()

                expect(countyScraperMap.countyA.mock.calls).toEqual([
                    ['ab cd'],
                    ['de fg'],
                    ['ifsingle donotread'],
                ])
                expect(countyScraperMap.countyB.mock.calls).toEqual([
                    ['ab cd'],
                    ['de fg'],
                    ['ifsingle donotread'],
                ])
                expect(lib.io.appendOutputData).toBeCalledTimes(2)
            })
        })
    })
})
