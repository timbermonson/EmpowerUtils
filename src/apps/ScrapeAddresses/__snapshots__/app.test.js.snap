// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Address Scraper App > getOutputText() > Both output > Handles 1 name 1`] = `"{"george washington":{"fullName":"georg washton","addressList":[{"street":" pick this ","city":" city"},{"street":" don't pick this ","city":"bad city "}]}}	george washington	 pick this ,  city |  don't pick this , bad city "`;

exports[`Address Scraper App > getOutputText() > Both output > Handles 2 names 1`] = `"{"george washington":{"fullName":"georg washton","addressList":[{"street":" pick this ","city":" city"},{"street":" don't pick this ","city":"bad city "}]},"john smith":{"fullName":"johnathan smithson","addressList":[{"street":" jpick this ","city":" jcity"},{"street":" jdon't pick this ","city":"jbad city "}]}}	george washington	 pick this ,  city |  don't pick this , bad city 	john smith	 jpick this ,  jcity |  jdon't pick this , jbad city "`;

exports[`Address Scraper App > getOutputText() > Excel output > Handles 1 name 1`] = `"george washington	 pick this ,  city |  don't pick this , bad city "`;

exports[`Address Scraper App > getOutputText() > Excel output > Handles 2 names 1`] = `"george washington	 pick this ,  city |  don't pick this , bad city 	john smith	 jpick this ,  jcity |  jdon't pick this , jbad city "`;

exports[`Address Scraper App > getOutputText() > JSON output > Handles 1 name 1`] = `"{"george washington":{"fullName":"georg washton","addressList":[{"street":" pick this ","city":" city"},{"street":" don't pick this ","city":"bad city "}]}}"`;

exports[`Address Scraper App > getOutputText() > JSON output > Handles 2 names 1`] = `"{"george washington":{"fullName":"georg washton","addressList":[{"street":" pick this ","city":" city"},{"street":" don't pick this ","city":"bad city "}]},"john smith":{"fullName":"johnathan smithson","addressList":[{"street":" jpick this ","city":" jcity"},{"street":" jdon't pick this ","city":"jbad city "}]}}"`;

exports[`Address Scraper App > getSearchresultMapByName() > Handles 1 names 1`] = `
[
  [
    {
      "countyA": {
        "george washington": {
          "fullName": "george washington",
        },
      },
      "countyB": {
        "george washington": {
          "fullName": "george washington",
        },
      },
    },
  ],
]
`;

exports[`Address Scraper App > getSearchresultMapByName() > Handles 2 names 1`] = `
[
  [
    {
      "countyA": {
        "george washington": {
          "fullName": "george washington",
        },
        "john smith": {
          "fullName": "john smith",
        },
      },
      "countyB": {
        "george washington": {
          "fullName": "george washington",
        },
        "john smith": {
          "fullName": "john smith",
        },
      },
    },
  ],
]
`;

exports[`Address Scraper App > run() > Single Line > Handles 2 names, tries to format properly 1`] = `
[
  [
    "a	Astreet, Acity	b	Bstreet, Bcity
",
  ],
]
`;

exports[`Address Scraper App > run() > Single Line > Handles 2 names, uses formatter 1`] = `
[
  [
    "a	Astreet, Acity	b	Bstreet, Bcity
",
  ],
]
`;
